src/ui/public/vis_maps/heatmap.js

```diff
@@ -0,0 +1,200 @@
+import L from 'leaflet';
+import _ from 'lodash';
+import d3 from 'd3';
+import { EventEmitter } from 'events';
+
+/**
+ * Map overlay: canvas layer with leaflet.heat plugin
+ *
+ * @param map {Leaflet Object}
+ * @param geoJson {geoJson Object}
+ * @param params {Object}
+ */
+export default class Heatmap extends EventEmitter {
```

Comment:
The empty legend is still showing up, when it isn't in 5.2.2. Obviously, not a blocker, but just FYI.